### PR TITLE
Metrics utils

### PR DESCRIPTION
**Pull request by c-cube**

### CODE REVIEW

File: src/opentelemetry.ml [hunk 1]
```diff
@@ -658,6 +705,32 @@ module Metrics = struct
     Collector.send_metrics [rm] ~ret:ignore
 end
 
+(** A set of callbacks that produce metrics when called.
+
+    The metrics are automatically called regularly.
+
+    This allows applications to register metrics callbacks from various points
+    in the program (or even in librariese), and not worry about setting
```
Comment (mattjbray) [inline src/opentelemetry.ml:713]:
```suggestion
    in the program (or even in libraries), and not worry about setting
```

File: src/client/opentelemetry_client_ocurl.ml [hunk 2]
```diff
@@ -200,6 +201,9 @@ let mk_emitter ~(config:Config.t) () : (module EMITTER) =
 
   let ((module C) as curl) = (module Curl() : CURL) in
 
+  let on_tick_cbs_ = ref (ref []) in
```
Comment (mattjbray) [inline src/client/opentelemetry_client_ocurl.ml:204]: Haven't quite understood why this is a double `ref`, but I'm sure there's a good reason :)

**Review by mattjbray — APPROVED** [review]
👍

File: src/client/opentelemetry_client_ocurl.ml (hunk 2 above)
Comment (c-cube) [reply src/client/opentelemetry_client_ocurl.ml:204]: for initialization reasons, yeah…

**Pull request merged**